ekncontent/ekncontent/eknc-domain.c

```diff
@@ -730,44 +730,69 @@ eknc_domain_get_fixed_query_finish (EkncDomain *self,
 typedef struct
 {
   // List of EkncContentModel items
-  GSList *models;
+  GPtrArray *models;
   guint total_models;
   guint upper_bound;
+  guint num_received;
 } QueryState;
 
 static void
 query_state_free (gpointer data)
 {
   QueryState *state = data;
-  g_slist_free_full (state->models, g_object_unref);
+  g_clear_pointer (&state->models, g_ptr_array_unref);
```

Comment:
I think that's right. The memory ownership semantics around this point got a little confusing - I initially tried having the pointer array reduce the refcount on the models when it was destroyed though that caused it to free memory that was being used elsewhere. Would it make more sense to take a ref on each of the objects when marshalling them into the GList and then use `g_ptr_array_new_full` above?